docs/topics/xml-rule-when-syntax.adoc

```diff
@@ -445,3 +447,91 @@ from="MyEjbRule"
 ----
 
 |====
+
+[[has-hint-syntax]]
+== <has-hint> Syntax
+
+=== Summary
+
+Use the `<has-hint>` element to test whether a file or line has a hint already associated with it. It is primarily used to prevent adding classification to a file if a hint already exists, or to implement rules for default execution when no other conditions apply. For a better understanding of the `<has-hint>` condition, see the JavaDoc for the http://windup.github.io/windup/docs/latest/javadoc/org/jboss/windup/reporting/config/HasHint.html[HasHint] class.
+
+The following is an example of a rule that checks to see if a hint exists for an IBM JMS destination message, and if not includes it.
+
+[source,options="nowrap"]
+----
+<rule id="websphere-jms-eap7-03000">
+  <when>
+    <javaclass references="{package}.{prefix}{type}Message" />
+  </when>
+  <perform>
+    <iteration>
+      <when>
+        <not>
+          <has-hint />
+        </not>
+      </when>
+      <perform>
+        <hint title="IBM JMS destination message" effort="1" category-id="mandatory">
+          <message>
+            JMS `{package}.{prefix}{type}Message` messages represent the actual data passed through JMS destinations. This reference should be
+            replaced with the Java EE standard API `javax.jms.{type}Message`.
+          </message>
+          <link href="https://docs.oracle.com/javaee/7/tutorial/jms-concepts003.htm#sthref2271" title="Java EE 7 JMS Tutorial - Message API" />
+          <tag>jms</tag>
+          <tag>websphere</tag>
+        </hint>
+      </perform>
+    </iteration>
+  </perform>
+  <where param="type">
+    <matches pattern="(Text|Stream|Object|Map|Bytes)?" />
+  </where>
+  <where param="prefix">
+    <matches pattern="(JMS|MQe|MQ)" />
+  </where>
+  <where param="package">
+    <matches pattern="com.ibm(\..*)?\.jms" />
+  </where>
+</rule>
+----
+
+=== Construct a <has-hint>
+
+The `<has-hint>` element is used to determine if a hint exists for a file or line. It does not have any child elements.
+
+==== <has-hint> Element Attributes
+
+[cols="1,1,3", options="header"]
+|====
+|Attribute Name
+|Type
+|Description
+
+|message
+|String
+|An optional argument allowing you to match the hint against the provided message string.
+|====
+
+[[has-classification-syntax]]
+== <has-classification> Syntax
+
+=== Summary
+
+Use the `<has-classification>` element to test whether a file or line has a given classification. For a better understanding of the `<has-classification>` condition, see the JavaDoc for the http://windup.github.io/windup/docs/latest/javadoc/org/jboss/windup/reporting/config/HasClassification.html[HasClassification] class. 
```

Comment:
I think `given` can be removed from this line.
Also adding the same detailed description that you wrote for the `<has-hint>` will definitely enhance this description.

Reply:
`given` has been removed from these sentences.

I've included the extra description for `has-classification`. Are these used for the same purposes, just one checking for hints and the other for classifications? Essentially, are both used to implement rules for default execution?

Reply:
_"Are these used for the same purposes, just one checking for hints and the other for classifications?"_ Yes, exactly
_"Essentially, are both used to implement rules for default execution?"_ Essentialy, yes